File: disks.py
def min_disks(file_sizes, disk_capacity):
    disks = 1
    current_disk_space = disk_capacity

    for size in file_sizes:
        if size > current_disk_space:
            disks += 1
            current_disk_space = disk_capacity
        current_disk_space -= size

    return disks

# Example usage
file_sizes = [100, 200, 300, 400, 500]
disk_capacity = 700

result = min_disks(file_sizes, disk_capacity)
print(f"Minimum number of disks required: {result}")

File: disks.c
#include <stdio.h>
#include <stdlib.h>

int minDisks(int* fileSizes, int numFiles, int diskCapacity) {
    int disks = 1;
    int currentDiskSpace = diskCapacity;

    for (int i = 0; i < numFiles; i++) {
        if (fileSizes[i] > currentDiskSpace) {
            disks++;
            currentDiskSpace = diskCapacity;
        }
        currentDiskSpace -= fileSizes[i];
    }

    return disks;
}

int main() {
    int fileSizes[] = {100, 200, 300, 400, 500};
    int numFiles = sizeof(fileSizes) / sizeof(fileSizes[0]);
    int diskCapacity = 700;

    int result = minDisks(fileSizes, numFiles, diskCapacity);
    printf("Minimum number of disks required: %d\n", result);

    return 0;
}
